src/opendr/perception/object_tracking_3d/single_object_tracking/vpit/vpit_object_tracking_3d_learner.py

```diff
@@ -0,0 +1,1593 @@
+            checkpoint_after_iter=self.checkpoint_after_iter,
+            checkpoints_path=checkpoints_path,
+            display_step=display_step,
+            device=self.device,
+            image_shape=image_shape,
+            evaluate=evaluate,
+            context_amount=self.context_amount,
+            target_size=self.target_size,
+            search_size=self.search_size,
+            debug=debug,
+            train_steps=steps,
+            loss_function=self.loss_function,
+            r_pos=self.r_pos,
+            infer_point_cloud_mapper=self.infer_point_cloud_mapper,
+            augment=self.augment,
+            augment_rotation=self.augment_rotation,
+            search_type=self.search_type,
+            target_type=self.target_type,
+            training_method=self.training_method,
+            train_pseudo_image=self.train_pseudo_image,
+            regress_vertical_position=self.regress_vertical_position,
+            regression_training_isolation=self.regression_training_isolation,
+            overwrite_strides=self.overwrite_strides,
+            upscaling_mode=self.upscaling_mode,
+            bof_training_steps=0 if self.bof_mode == "none" else self.bof_training_steps,
+        )
+
+        logger.close()
+
+    def eval(
+        self,
+        dataset,
+        ground_truth_annotations=None,
+        logging_path=None,
+        silent=False,
+        verbose=False,
+        image_shape=(370, 1224),
+        count=None,
+    ):
+
+        logger = Logger(silent, verbose, logging_path)
+
+        (_, eval_dataset_iterator, ground_truth_annotations,) = self.__prepare_datasets(
+            None,
+            dataset,
+            self.input_config,
+            self.evaluation_input_config,
+            self.model_config,
+            self.voxel_generator,
+            self.target_assigner,
+            ground_truth_annotations,
+            require_dataset=False,
+        )
+
+        result = evaluate(
+            self.siamese,
+            self.evaluation_input_config,
+            self.model_config,
+            self.mixed_optimizer,
+            self.model_dir,
+            self.float_dtype,
+            self.class_names,
+            self.center_limit_range,
+            eval_dataset_iterator=eval_dataset_iterator,
+            gt_annos=ground_truth_annotations,
+            predict_test=False,
+            log=logger.log,
+            device=self.device,
+            image_shape=image_shape,
+            count=count,
+        )
+
+        logger.close()
+
+        return result
+
+    def create_pseudo_image(self, point_clouds, pc_range):
+
+        if self.model is None:
+            raise ValueError("No model loaded or created")
+
+        result = infer_create_pseudo_image(
+            self.model.branch,
+            point_clouds,
+            pc_range,
+            self.infer_point_cloud_mapper,
+            self.float_dtype,
+            times=self.times,
+        )
+
+        return result
```

Comment:
This method returns here, rest of the code is unreachable.

Reply:
Updated